### PR TITLE
Allow specifying environment variables for Terraform commands

### LINKED ISSUE #2770: Allow configuring environment variables for Terraform command used while deploying Terraform application
**What would you like to be added**:

**Why is this needed**:

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2770

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 1]
```diff
@@ -315,6 +315,7 @@ One of `yamlField` or `regex` is required.
 | vars | []string | List of variables that will be set directly on terraform commands with `-var` flag. The variable must be formatted by `key=value`. | No |
 | varFiles | []string | List of variable files that will be set on terraform commands with `-var-file` flag. | No |
 | commandFlags | [TerraformCommandFlags](#terraformcommandflags) | List of additional flags will be used while execute terraform commands. | No |
+| commandEnvs | [TerraformCommandEnvs](#terraformcommandenvs) | List of additional environment variables will be used while execute terraform commands. | No |
```
Comment (ono-max) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:318]: nit
```suggestion
| commandEnvs | [TerraformCommandEnvs](#terraformcommandenvs) | List of additional environment variables will be used while executing terraform commands. | No |
```

File: pkg/config/deployment_terraform.go [hunk 2]
```diff
@@ -51,6 +51,8 @@ type TerraformDeploymentInput struct {
 	AutoRollback bool `json:"autoRollback"`
 	// List of additional flags will be used while execute terraform commands.
 	CommandFlags TerraformCommandFlags `json:"commandFlags"`
+	// List of additional environment variables will be used while execute terraform commands.
```
Comment (ono-max) [inline pkg/config/deployment_terraform.go:54]:
```suggestion
	// List of additional environment variables will be used while executing terraform commands.
```

**Comment by ono-max** [comment]
Nice.

/lgtm

**Comment by nakabonne** [comment]
I'm sure it will work as well as we'd like 👍 
/approve

**Pull request merged**